src/pages/api/modules.ts

```diff
@@ -0,0 +1,31 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import data from "../../../data.json";
+
+const modules = (req: NextApiRequest, res: NextApiResponse) => {
+  const searchTerm = req.query.search;
+
+  if (!searchTerm) {
+    res.statusCode = 200;
+    return res.json(data);
+  }
+
+  if (Array.isArray(searchTerm)) {
```

Comment:
What scenario would cause `searchTerm` to be an array? Is that a Next.js thing?

Reply:
Yup, typing of `req.query.search` is `string | string[] | undefined` as you can provide a query param multiple times